### ISSUE TITLE
[Pimatic-Max] Failed to start Pimatic

### ISSUE THREAD

**Issue by bonkersGER** [pimatic/pimatic#1025]
```
20:01:32.951 [pimatic] A uncaught exception occured: TypeError: must start with number, buffer, array or string
20:01:32.951 [pimatic]>    at fromObject (buffer.js:138:11)
20:01:32.951 [pimatic]>    at new Buffer (buffer.js:65:10)
20:01:32.951 [pimatic]>    at MaxCube.parseCommandMetadata (/home/pi/pimatic-app/node_modules/pimatic-max/node_modules/max-control/index.j                                                   s:180:24)
20:01:32.951 [pimatic]>    at MaxCube.parseCommand (/home/pi/pimatic-app/node_modules/pimatic-max/node_modules/max-control/index.js:117:19                                                   )
20:01:32.951 [pimatic]>    at MaxCube.<anonymous> (/home/pi/pimatic-app/node_modules/pimatic-max/node_modules/max-control/index.js:75:26)
20:01:32.951 [pimatic]>    at Array.forEach (native)
20:01:32.951 [pimatic]>    at MaxCube.onData (/home/pi/pimatic-app/node_modules/pimatic-max/node_modules/max-control/index.js:70:8)
20:01:32.951 [pimatic]>    at emitOne (events.js:77:13)
20:01:32.951 [pimatic]>    at Socket.emit (events.js:169:7)
20:01:32.951 [pimatic]>    at Socket.emit (/home/pi/pimatic-app/node_modules/pimatic/startup.coffee:63:24)
20:01:32.951 [pimatic]>    at readableAddChunk (_stream_readable.js:153:18)
20:01:32.951 [pimatic]>    at Socket.Readable.push (_stream_readable.js:111:10)
20:01:32.951 [pimatic]>    at TCP.onread (net.js:531:20)
20:01:32.951 [pimatic]> This is most probably a bug in pimatic or in a module, please report it!

```

**Comment by davidh2k** [comment]
```
21:16:04.757 2016-11-29 Tuesday
21:16:11.293 [pimatic] Starting pimatic version 0.9.37
21:16:14.097 [pimatic, ppm] 2.15.11
21:16:14.164 [pimatic] Loading plugin: "pimatic-cron" (0.8.8)
21:16:14.753 [pimatic] Loading plugin: "pimatic-mobile-frontend" (0.9.11)
21:16:15.647 [pimatic] Loading plugin: "pimatic-max" (0.8.19)
21:16:15.742 [pimatic] Loading plugin: "pimatic-hap" (0.9.1)
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
21:16:16.440 [pimatic-cron] the time is: Tue Nov 29 2016 21:16:16 GMT+0100 (CET)
21:16:16.690 [pimatic-hap] Starting homekit bridge
21:16:16.896 [pimatic-max] Connected, waiting for first update from cube
21:16:16.951 [pimatic] New device "Kueche"...
21:16:16.956 [pimatic] The device type MaxHeatingThermostat does not implement a destroy function
21:16:17.345 [pimatic-max] got update { '16a6be':
21:16:17.345 [pimatic-max]>   { type: 'Heating Thermostat',
21:16:17.345 [pimatic-max]>     address: '16a6be',
21:16:17.345 [pimatic-max]>     serial: 'NEQ12525',
21:16:17.345 [pimatic-max]>     name: 'Thermostat-1',
21:16:17.345 [pimatic-max]>     roomId: 1 },
21:16:17.345 [pimatic-max]>  '16c4cd':
21:16:17.345 [pimatic-max]>   { type: 'Wall mounted Thermostat',
21:16:17.345 [pimatic-max]>     address: '16c4cd',
21:16:17.345 [pimatic-max]>     serial: 'NEQ11601',
21:16:17.345 [pimatic-max]>     name: 'WZ-Wandthermostat-1',
21:16:17.345 [pimatic-max]>     roomId: 1 },
21:16:17.345 [pimatic-max]>  '16a6c2':
21:16:17.345 [pimatic-max]>   { type: 'Heating Thermostat',
21:16:17.345 [pimatic-max]>     address: '16a6c2',
21:16:17.345 [pimatic-max]>     serial: 'NEQ12525',
21:16:17.345 [pimatic-max]>     name: 'KuechenThermostat-1',
21:16:17.345 [pimatic-max]>     roomId: 2 },
21:16:17.345 [pimatic-max]>  '16a6b1':
21:16:17.345 [pimatic-max]>   { type: 'Heating Thermostat',
21:16:17.345 [pimatic-max]>     address: '16a6b1',
21:16:17.345 [pimatic-max]>     serial: 'NEQ12525',
21:16:17.345 [pimatic-max]>     name: 'Bad-Thermostat-1',
21:16:17.345 [pimatic-max]>     roomId: 3 } }
21:16:17.397 [pimatic] A uncaught exception occured: Error: unsupported mode undefined
21:16:17.397 [pimatic]>    at /home/pimatic/pimatic-app/node_modules/pimatic-hap/accessories/thermostat.coffee:80:28
21:16:17.397 [pimatic]>    at MaxHeatingThermostat.<anonymous> (/home/pimatic/pimatic-app/node_modules/pimatic-hap/accessories/thermostat.coffee:101:13)
21:16:17.397 [pimatic]>    at emitOne (events.js:82:20)
21:16:17.397 [pimatic]>    at MaxHeatingThermostat.emit (events.js:169:7)
21:16:17.397 [pimatic]>    at MaxHeatingThermostat.HeatingThermostat._setMode (/home/pimatic/pimatic-app/node_modules/pimatic/lib/devices.coffee:576:8)
21:16:17.397 [pimatic]>    at MaxCube.<anonymous> (/home/pimatic/pimatic-app/node_modules/pimatic-max/maxthermostat.coffee:122:16)
21:16:17.397 [pimatic]>    at emitOne (events.js:82:20)
21:16:17.397 [pimatic]>    at MaxCube.emit (events.js:169:7)
21:16:17.397 [pimatic]>    at MaxCube.onData (/home/pimatic/pimatic-app/node_modules/pimatic-max/node_modules/max-control/index.js:79:8)
21:16:17.397 [pimatic]>    at emitOne (events.js:77:13)
21:16:17.397 [pimatic]>    at Socket.emit (events.js:169:7)
21:16:17.397 [pimatic]>    at Socket.emit (/home/pimatic/pimatic-app/node_modules/pimatic/startup.coffee:63:24)
21:16:17.397 [pimatic]>    at readableAddChunk (_stream_readable.js:153:18)
21:16:17.397 [pimatic]>    at Socket.Readable.push (_stream_readable.js:111:10)
21:16:17.397 [pimatic]>    at TCP.onread (net.js:537:20)
21:16:17.397 [pimatic]> This is most probably a bug in pimatic or in a module, please report it!
21:16:17.411 [pimatic] exiting...

```

**Comment by sweetpi** [comment]
I think is a problem with the pimatic-hap plugin. It seems it can't handle the case that the mode is undefined. 

Please report the problem at https://github.com/michbeck100/pimatic-hap 

You can try to start pimatic without the pimatic-hap event. Set a mode for all thermostats (run it a while, so that all changes are set) and then restart pimatic with pimatic-hap enabled.

I'm closing this here, because I think its an issue with pimatic-hap.

**Comment by michbeck100** [comment]
Ehoud be fixed with 0.9.2 of pimatic-hap. I've added a check for undefined. But i think this should also be fixed on pimatic-max side: There should never be a thermostat mode of undefined, not even after restart.

**Comment by mwittig** [comment]
I assume this has been fixed. Please re-open or open a new issue if these errors still occur.
